libevmasm/Assembly.cpp

```diff
@@ -1553,6 +1553,14 @@ LinkerObject const& Assembly::assembleEOF() const
 			}
 		}
 
+		if (ret.bytecode.size() - sectionStart > std::numeric_limits<uint16_t>::max())
+			// TODO: Include source location. Note that origin locations we have in debug data are
+			// not usable for error reporting when compiling pure Yul because they point at the optimized source.
+			throw Error(
+				2202_error,
+				Error::Type::CodeGenerationError,
+				"Code section too large for EOF."
+			);
```

Comment:
@rodiazet Once this PR gets merged, this is how EOF errors triggerable by user code should be reported.

The analyzer checks for the number of subs and functions should be replaced with this.